src/platform/guimac.mm

```diff
@@ -1409,6 +1409,7 @@ - (NSApplicationTerminateReply)applicationShouldTerminate:(NSApplication *)sende
 
 @implementation SSApplicationDelegate
 - (IBAction)preferences:(id)sender {
+    SolveSpace::SS.GW.MenuView(SolveSpace::Command::SHOW_TEXT_WND);
```

Comment:
This toggles the text window visibility.

Reply:
Oops, good to know haha, I will add a check around it to see if it's actually off

Reply:
Just do `SS.GW.showTextWindow = true;`

Reply:
But that would not update the checkmark if it's open or not, correct? e.g.:

```
    if(SS.TW.window) SS.TW.window->SetVisible(SS.GW.showTextWindow);
    showTextWndMenuItem->SetActive(SS.GW.showTextWindow);
```

Reply:
Oh yes, the horrible `EnsureValidActives` function. Go ahead with your plan.